monkey/monkey_island/cc/resources/pba_file_upload.py

```diff
@@ -96,16 +120,23 @@ def delete(self, target_os):
         if self._is_target_os_supported(target_os):
             return Response(status=HTTPStatus.UNPROCESSABLE_ENTITY, mimetype="text/plain")
 
-        filename_path = (
-            PBA_LINUX_FILENAME_PATH if target_os == "PBAlinux" else PBA_WINDOWS_FILENAME_PATH
-        )
-        filename = ConfigService.get_config_value(filename_path)
-        if filename:
+        original_agent_configuration = self._agent_configuration_repository.get_configuration()
+        self._update_config(target_os, "")
+
+        if target_os == LINUX_PBA_TYPE:
+            filename = original_agent_configuration.custom_pbas.linux_filename
+        else:
```

Comment:
We could also raise an error if it's an unrecognized OS type

Reply:
This could all be improved, but I want to maximize the work-not-done since we'll be rewriting this as a payload.